### ISSUE TITLE
Uncaught TypeError: Cannot read property 'replace' of undefined

### ISSUE THREAD

**Issue by ole-jorgen** [h3imdall/ftp-remote-edit#173]
[Enter steps to reproduce:]

1. ...
2. ...

**Atom**: 1.24.0 x64
**Electron**: 1.6.16
**OS**: Microsoft Windows 10 Home
**Thrown From**: [ftp-remote-edit](https://github.com/h3imdall/ftp-remote-edit) package 0.12.11

### Stack Trace

Uncaught TypeError: Cannot read property 'replace' of undefined

```
At C:\Users\Ole\.atom\packages\ftp-remote-edit\lib\helper\format.js:52

TypeError: Cannot read property 'replace' of undefined
    at trailingslashit (/packages/ftp-remote-edit/lib/helper/format.js:52:16)
    at /packages/ftp-remote-edit/lib/ftp-remote-edit.js:582:28
    at FtpRemoteEdit.autoRevealActiveFile (/packages/ftp-remote-edit/lib/ftp-remote-edit.js:595:14)
    at /packages/ftp-remote-edit/lib/ftp-remote-edit.js:81:12
    at Function.module.exports.Emitter.simpleDispatch (~/AppData/Local/atom/app-1.24.0/resources/app/node_modules/event-kit/lib/emitter.js:25:20)
    at Emitter.module.exports.Emitter.emit (~/AppData/Local/atom/app-1.24.0/resources/app/node_modules/event-kit/lib/emitter.js:141:34)
    at Workspace.didActivatePaneContainer (~/AppData/Local/atom/app-1.24.0/resources/app/src/workspace.js:477:28)
    at WorkspaceCenter.paneContainer.onDidActivatePane (~/AppData/Local/atom/app-1.24.0/resources/app/src/workspace-center.js:12:59)
    at Function.module.exports.Emitter.simpleDispatch (~/AppData/Local/atom/app-1.24.0/resources/app/node_modules/event-kit/lib/emitter.js:25:20)
    at Emitter.module.exports.Emitter.emit (~/AppData/Local/atom/app-1.24.0/resources/app/node_modules/event-kit/lib/emitter.js:141:34)
    at PaneContainer.didActivatePane (~/AppData/Local/atom/app-1.24.0/resources/app/src/pane-container.js:264:24)
    at Pane.activate (~/AppData/Local/atom/app-1.24.0/resources/app/src/pane.js:1002:46)
    at Pane.focus (~/AppData/Local/atom/app-1.24.0/resources/app/src/pane.js:393:23)
    at /app.asar/src/pane-element.js:51:25)
```

### Commands

```
     -5:00.6.0 ftp-remote-edit:toggle (input.hidden-input)
     -4:59.4.0 core:confirm (input.hidden-input)
```

### Non-Core Packages

```
file-icons 2.1.17 
ftp-remote-edit 0.12.11 
teletype 0.9.0 
title-bar-replacer 1.8.1 
```

**Comment by peisia91** [comment]
[Enter steps to reproduce:]

1. ...
2. ...

**Atom**: 1.24.0 x64
**Electron**: 1.6.16
**OS**: Mac OS X 10.13.3
**Thrown From**: [ftp-remote-edit](https://github.com/h3imdall/ftp-remote-edit) package 0.12.11


### Stack Trace

Uncaught TypeError: Cannot read property 'replace' of undefined

```
At /Users/toydream/.atom/packages/ftp-remote-edit/lib/helper/format.js:52

TypeError: Cannot read property 'replace' of undefined
    at trailingslashit (/packages/ftp-remote-edit/lib/helper/format.js:52:16)
    at /packages/ftp-remote-edit/lib/ftp-remote-edit.js:582:28
    at FtpRemoteEdit.autoRevealActiveFile (/packages/ftp-remote-edit/lib/ftp-remote-edit.js:595:14)
    at /packages/ftp-remote-edit/lib/ftp-remote-edit.js:81:12
    at Function.module.exports.Emitter.simpleDispatch (/Applications/Atom.app/Contents/Resources/app/node_modules/event-kit/lib/emitter.js:25:20)
    at Emitter.module.exports.Emitter.emit (/Applications/Atom.app/Contents/Resources/app/node_modules/event-kit/lib/emitter.js:141:34)
    at Workspace.didActivatePaneContainer (/Applications/Atom.app/Contents/Resources/app/src/workspace.js:477:28)
    at WorkspaceCenter.paneContainer.onDidActivatePane (/Applications/Atom.app/Contents/Resources/app/src/workspace-center.js:12:59)
    at Function.module.exports.Emitter.simpleDispatch (/Applications/Atom.app/Contents/Resources/app/node_modules/event-kit/lib/emitter.js:25:20)
    at Emitter.module.exports.Emitter.emit (/Applications/Atom.app/Contents/Resources/app/node_modules/event-kit/lib/emitter.js:141:34)
    at PaneContainer.didActivatePane (/Applications/Atom.app/Contents/Resources/app/src/pane-container.js:264:24)
    at Pane.activate (/Applications/Atom.app/Contents/Resources/app/src/pane.js:1002:46)
    at Pane.focus (/Applications/Atom.app/Contents/Resources/app/src/pane.js:393:23)
    at /app.asar/src/pane-element.js:51:25)
```

### Commands

```
     -3:57.2.0 ftp-remote-edit:toggle (atom-pane.pane.active)
     -3:44.9.0 ftp-remote-edit:edit-servers (atom-workspace.workspace.scrollbars-visible-always.theme-atom-light-syntax.theme-one-light-ui)
  5x -3:27 core:backspace (input.hidden-input)
     -3:23.5.0 core:move-down (input.hidden-input)
  5x -3:23.3.0 core:move-left (input.hidden-input)
     -3:07.1.0 core:paste (input.hidden-input)
     -3:06.3.0 core:move-left (input.hidden-input)
     -3:05.9.0 core:backspace (input.hidden-input)
     -3:02.2.0 core:move-right (input.hidden-input)
     -3:02.0 core:delete (input.hidden-input)
     -2:48.7.0 core:backspace (input.hidden-input)
     -2:13.1.0 core:paste (input.hidden-input)
     -1:59.8.0 nuclide-settings:toggle (ol.ftp-remote-edit-list.full-menu.list-tree.has-collapsable-children.focusable-panel)
  2x -1:03.0 ftp-remote-edit:edit-servers (ol.ftp-remote-edit-list.full-menu.list-tree.has-collapsable-children.focusable-panel)
```

### Non-Core Packages

```
file-icons 2.1.17 
ftp-remote-edit 0.12.11 
highlight-selected 0.13.1 
language-babel 2.84.0 
language-graphql 0.9.0 
language-haskell 1.17.3 
language-ini 1.19.0 
language-lua 0.9.11 
language-ocaml 1.9.5 
language-rust 0.4.12 
language-scala 1.1.9 
language-swift 0.5.0 
language-thrift 1.0.2 
MagicPython 1.0.12 
nuclide 0.286.0 
nuclide-format-js 0.0.44 
set-syntax 0.4.0 
sort-lines 0.18.0 
tool-bar 1.1.6 
```

**Comment by cschindl** [comment]
Hi,

the error appears to be in the "Auto Reveal Active File" function. I'll look into the problem. In the meantime, you can disable this function in the settings under the menu item "Tree View" - "Auto Reveal Active File"

**Comment by cschindl** [comment]
Fixed in 6ac22ae04373be472edeb08d15ccf993efb229cb

**Comment by agentram** [comment]
[Enter steps to reproduce:]

1. ...
2. ...

**Atom**: 1.24.0 x64
**Electron**: 1.8.2
**OS**: "Manjaro Linux"
**Thrown From**: [ftp-remote-edit](https://github.com/h3imdall/ftp-remote-edit) package 0.12.11


### Stack Trace

Uncaught TypeError: Cannot read property 'replace' of undefined

```
At /home/oleg/.atom/packages/ftp-remote-edit/lib/helper/format.js:52

TypeError: Cannot read property 'replace' of undefined
    at trailingslashit (/packages/ftp-remote-edit/lib/helper/format.js:52:16)
    at /packages/ftp-remote-edit/lib/ftp-remote-edit.js:582:28
    at FtpRemoteEdit.autoRevealActiveFile (/packages/ftp-remote-edit/lib/ftp-remote-edit.js:595:14)
    at /packages/ftp-remote-edit/lib/ftp-remote-edit.js:81:12
    at Function.module.exports.Emitter.simpleDispatch (/usr/lib/atom/node_modules/event-kit/lib/emitter.js:25:14)
    at Emitter.module.exports.Emitter.emit (/usr/lib/atom/node_modules/event-kit/lib/emitter.js:141:28)
    at Workspace.didChangeActivePaneItemOnPaneContainer (/build/atom/src/atom-1.24.0/out/app/src/workspace.js:426:20)
    at WorkspaceCenter.paneContainer.onDidChangeActivePaneItem (/usr/lib/atom/src/workspace-center.js:17:14)
    at Function.module.exports.Emitter.simpleDispatch (/usr/lib/atom/node_modules/event-kit/lib/emitter.js:25:14)
    at Emitter.module.exports.Emitter.emit (/usr/lib/atom/node_modules/event-kit/lib/emitter.js:141:28)
    at PaneContainer.didChangeActiveItemOnPane (/usr/lib/atom/src/pane-container.js:280:20)
    at PaneContainer.didActivatePane (/usr/lib/atom/src/pane-container.js:258:12)
    at Pane.activate (/usr/lib/atom/src/pane.js:998:40)
    at Pane.focus (/usr/lib/atom/src/pane.js:389:17)
    at HTMLElement.<anonymous> (/usr/lib/atom/src/pane-element.js:51:25)
```

### Commands

```
     -3:24.4.0 ftp-remote-edit:edit-servers (span.name.icon.icon-file-symlink-directory.icon-sync.spin)
     -3:21.8.0 core:backspace (input.hidden-input)
     -3:03.0 intentions:highlight (input.hidden-input)
     -3:02.7.0 core:save (input.hidden-input)
     -2:55.8.0 script:close-view (atom-workspace.workspace.scrollbars-visible-always.theme-city-lights-syntax.theme-city-lights-ui)
     -2:49 intentions:highlight (input.hidden-input)
     -2:48.9.0 ftp-remote-edit:toggle (input.hidden-input)
     -1:22.9.0 intentions:highlight (input.hidden-input)
     -1:22.8.0 ftp-remote-edit:toggle (input.hidden-input)
     -1:21.7.0 intentions:highlight (input.hidden-input)
     -1:21.6.0 ftp-remote-edit:toggle (input.hidden-input)
     -1:18.2.0 intentions:highlight (input.hidden-input)
     -1:18 core:save (input.hidden-input)
     -1:13.9.0 core:backspace (input.hidden-input)
     -1:13.1.0 intentions:highlight (input.hidden-input)
     -1:13 core:save (input.hidden-input)
```

### Non-Core Packages

```
atom-beautify 0.32.2 
autocomplete-bash-builtins 0.3.5 
autocomplete-python 1.10.5 
busy-signal 1.4.3 
city-lights-icons 1.1.1 
city-lights-syntax 1.1.8 
city-lights-ui 1.3.3 
ftp-remote-edit 0.12.11 
highlight-selected 0.13.1 
intentions 1.1.5 
linter 2.2.0 
linter-checkbashisms 0.1.5 
linter-python 3.1.2 
linter-ui-default 1.7.1 
script 3.17.3 
```

**Comment by FabrizioCaldarelli** [comment]
The issue is solved with commit c47b005464f745c205b8740e244f4a645d18b314 but it is development branch, so it is not available for master branch.

Infact if you try to

- Toggle ftp remote package;
- Open a file in a panel;
- Then click on menu File -> New file;
- Try to change from remote file panel and new file panel this will show the problem;

Can this commit be available in master?

**Comment by cschindl** [comment]
Hi @FabrizioCaldarelli,

I will apply the changes to the master and ask @h3imdall to publish a new release.
